File: app/src/main/java/com/example/radioplayer/corejava/GetRadioStationsByCountryCodeUseCase.java
package com.example.radioplayer.corejava;


import com.example.radioplayer.domainjava.RadioStationsResponse;
import com.example.radioplayer.infrajava.RadioStationRepo;

import javax.inject.Inject;

import retrofit2.Call;

/**
 * Get radio stations by a specific country code usecase.
 */
public class GetRadioStationsByCountryCodeUseCase {

    private final RadioStationRepo radioStationRepo;

    public GetRadioStationsByCountryCodeUseCase(RadioStationRepo radioStationRepo) {
        this.radioStationRepo = radioStationRepo;
    }

    public Call<RadioStationsResponse> execute(int country) {
        return radioStationRepo.getStationsByCountryCode(country);
    }
}

File: app/src/main/java/com/example/radioplayer/core/GetRadioStationsByCountryCodeUseCase.kt
//package com.example.radioplayer.core
//
//import com.example.radioplayer.domain.RadioStationsResponse
//import com.example.radioplayer.infra.RadioStationRepo
//import retrofit2.Response
//import javax.inject.Inject
//
///** Get radio stations by a specific country code usecase */
//class GetRadioStationsByCountryCodeUseCase @Inject constructor(private val radioStationRepo: RadioStationRepo) {
//    suspend operator fun invoke(
//        country: Int,
//    ): Response<RadioStationsResponse> {
//        return radioStationRepo.getStationsByCountryCode(country)
//    }
//}